src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -89,42 +76,42 @@ namespace {
 WaveClipAdjustBorderHandle::AdjustPolicy::~AdjustPolicy() = default;
 
 namespace {
-double GetLeftAdjustLimit(
-   const WaveClip& clip, const WaveTrack& track, bool adjustingLeftBorder,
-   bool isStretchMode)
+double GetLeftAdjustLimit(const WaveTrack::Interval& interval,
+                          const WaveTrack& track,
+                          bool adjustingLeftBorder,
+                          bool isStretchMode)
 {
    if (!adjustingLeftBorder)
       // A clip of one sample is invisible - who needs that ? Leave at least 2.
-      return clip.GetPlayStartTime() + 2.0 / clip.GetRate();
-
-   const auto prevClip = track.GetNextClip(clip, PlaybackDirection::backward);
-   if (isStretchMode)
-      return prevClip ? prevClip->GetPlayEndTime() :
-                        std::numeric_limits<double>::lowest();
-   else
-      return prevClip ?
-                std::max(
-                   clip.GetSequenceStartTime(), prevClip->GetPlayEndTime()) :
-                clip.GetSequenceStartTime();
+      return interval.Start() + 2.0 / track.GetRate();
+
+   const auto prevInterval = track.GetNextInterval(interval, PlaybackDirection::backward);
+   if(isStretchMode)
+      return prevInterval ? prevInterval->End() :
+                            std::numeric_limits<double>::lowest();
+   if(prevInterval)
+      return std::max(interval.GetClip(0)->GetSequenceStartTime(),
+                prevInterval->End());
+   return interval.GetClip(0)->GetSequenceStartTime();
 }
 
 double GetRightAdjustLimit(
-   const WaveClip& clip, const WaveTrack& track, bool adjustingLeftBorder,
+   const WaveTrack::Interval& interval, const WaveTrack& track, bool adjustingLeftBorder,
    bool isStretchMode)
 {
    if (adjustingLeftBorder)
       // A clip of one sample is invisible - who needs that ? Leave at least 2.
-      return clip.GetPlayEndTime() - 2.0 / clip.GetRate();
+      return interval.End() - 2.0 / track.GetRate();
 
-   const auto nextClip = track.GetNextClip(clip, PlaybackDirection::forward);
+   const auto nextInterval = track.GetNextInterval(interval, PlaybackDirection::forward);
    if (isStretchMode)
-      return nextClip ? nextClip->GetPlayStartTime() :
-                        std::numeric_limits<double>::max();
-   else
-      return nextClip ?
-                std::min(
-                   clip.GetSequenceEndTime(), nextClip->GetPlayStartTime()) :
-                clip.GetSequenceEndTime();
+      return nextInterval ? nextInterval->Start() :
+                            std::numeric_limits<double>::max();
+
+   if(nextInterval)
+      return std::min(interval.GetClip(0)->GetSequenceEndTime(),
+                      nextInterval->Start());
+   return interval.GetClip(0)->GetSequenceEndTime();
```

Comment:
Later `GetClip` will not take an argument because clips will really be wide in stereo tracks.

New calls of that function with constant 0 argument are acceptable now.  I don’t want more calls with nonzero.